source/extensions/filters/network/redis_proxy/command_splitter_impl.cc

```diff
@@ -1311,7 +1311,13 @@ SplitRequestPtr ScanRequest::create(Router& router, Common::Redis::RespValuePtr&
   if (cursor.length() > 4) {
     std::string index = cursor.substr(cursor.length() - 4);
     cursor = cursor.substr(0, cursor.length() - 4);
-    shard_idx = std::stoi(index);
+    try {
+      shard_idx = std::stoi(index);
+    } catch (const std::invalid_argument& e) {
+      ENVOY_LOG(error, "Invalid shard index: '{}'", e.what());
+      callbacks_.onResponse(Common::Redis::Utility::makeError(
+                            fmt::format("unexpected or wrong shard index from cursor")));
+    }
```

Comment:
Add a return null 
have you tested this change ? without adding return it could crash